src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1392,9 +1396,13 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
 
                 safetensors_variant_filenames = {f for f in variant_filenames if f.endswith(".safetensors")}
                 safetensors_model_filenames = {f for f in model_filenames if f.endswith(".safetensors")}
+                # `not is_sharded` because sharded checkpoints with a variant
+                # ("fp16") for example may have lesser shards actually. Consider
+                # https://huggingface.co/fal/AuraFlow/tree/main/transformer, for example.
```

Comment:
so if you look at the `variant_compatible_siblings` method, https://github.com/huggingface/diffusers/blob/14f6464bef677e47e1ff13a12f4ddd97e7f3e973/src/diffusers/pipelines/pipeline_loading_utils.py#L186 
it returns 2 things, `usable_filenames` (the `model_filenames` here), which is basically `variant_filenames` plus any non-variant checkpoints that do not have a variant, so we are not actually comparing variant vs non-varaint here, and regardless it's shared or not, the model_filenmaes should not contain more files than variant_filenames

Reply:
https://github.com/huggingface/diffusers/pull/9253#issuecomment-2370236958